internal/beater/java_attacher/java_attacher.go

```diff
@@ -167,6 +177,71 @@ func (j *JavaAttacher) Run(ctx context.Context) error {
 	}
 }
 
+func (j *JavaAttacher) deleteTempDir(tempDir string) {
+	func(name string) {
+		err := os.RemoveAll(name)
+		if err != nil {
+			j.logger.Errorf("failed to delete Java attacher temp dir at %v: %v", tempDir, err)
+		}
+	}(tempDir)
+}
+
+// Creates a temp dir for the java attacher jar with the attacher jar in it, both with 0755 access permissions.
+// The assumption is that the default temp dir will be readable for all users.
+// However, there may be systems in which the temp directory is user-specific,
+// in which case we'll need to modify this function to run a temp dir search per user,
+// maybe by externally executing `echo XXX` and reading the output, relying on environment
+// variables as implemented in os.TempDir, or looking in known paths.
+// In such case, we would switch to using a cache of temp-agent-dir per user.
+func (j *JavaAttacher) createAttacherTempDir() (string, error) {
+	tempDir, err := os.MkdirTemp("", "elasticapmagent")
+	if err != nil {
+		return "", err
+	}
+	// todo: check if we need to use different value for specific OSs
+	// make sure this dir is readable and executable by all users and writable only by the current user
+	err = os.Chmod(tempDir, 0755)
+	if err != nil {
+		j.logger.Errorf("failed to change permissions for %v to make it readable for all users: %v", tempDir, err)
+	}
+	bundledAttacherFile, err := os.Open(bundledJavaAttacher)
+	if err != nil {
+		defer j.deleteTempDir(tempDir)
+		return "", fmt.Errorf("failed to open bundled attacher jar: %w", err)
+	}
+	defer j.closeFile(bundledAttacherFile)
+	tmpAttacherJarPath := tmpJavaAttacherJarPath(tempDir)
+	tmpAttacherJarFile, err := os.Create(tmpAttacherJarPath)
+	if err != nil {
+		defer j.deleteTempDir(tempDir)
+		return "", fmt.Errorf("failed to create tmp attacher jar: %w", err)
+	}
+	defer j.closeFile(tmpAttacherJarFile)
+	nBytes, err := io.Copy(tmpAttacherJarFile, bundledAttacherFile)
+	if err != nil {
+		defer j.deleteTempDir(tempDir)
+		return "", fmt.Errorf("failed to copy bundled attacher jar to %v: %w", bundledJavaAttacher, err)
+	}
+	j.logger.Debugf("%v (%v bytes) successfully copied to %v", bundledJavaAttacher, nBytes, tmpAttacherJarPath)
+	err = os.Chmod(tmpAttacherJarPath, 0755)
```

Comment:
I thought so to, but luckily I tested and apparently it does in some cases.

Reply:
Can you please elaborate? What are those cases?

Reply:
Amazon Linux, at least

Reply:
Please elaborate further: what happens when you use mode `0644` for the attacher jar on that platform?

Reply:
It failed with an access failure message. Trying with `0744` failed and with `0755` succeeded

Reply:
I'm trying to understand what is causing the failure, so please give all the details. What is the error message? Maybe if you can provide the steps you took to test it, I can see it for myself.

I'm having a hard time believing that the file needs to be executable. Did you perhaps set both the temp directory _and_ the file to 0644? If that were the case, then it would make sense. Directories do need to be executable.

Reply:
> Maybe if you can provide the steps you took to test it, I can see it for myself

I built the APM Server, replaced the binaries on my Linux machine within my Elastic Agent installation and restarted the agent, then restarted the test app. 

> Did you perhaps set both the temp directory and the file to 0644? 

Very likely! IIRC, I tried only when both were either `0744` or `0755`

>  Directories do need to be executable

I didn't know that! Makes even less sense to me than requiring the jar to executable, given that the actual executable is the java binary...

I will definitely try it out. I would have reproduced now, but the current version is 8.5-SNAPSHOT while I am testing with 8.4-SNAPSHOT and merging with 8.4 requires some effort. I'd be happy to know of an easy way to build as a different version. I will get to testing that later today, preferably after we discuss offline.

Reply:
I thought the comment above was specific to running `java` and pointing to the target jar, but it's about the directory that needs to be executable in Linux in order to be accessible in general, thanks for pointing it out @axw 🙏

Reply:
Modes are `0700` for the tmp dir and `0600` for the jar, and the owner is set to the application user